npm/parseSelector_test.go

```diff
@@ -488,41 +446,113 @@ func TestParseSelector(t *testing.T) {
 
 	*me = append(*me, doesNotExist)
 
-	labels, keys, vals = parseSelector(selector)
+	labels, vals = parseSelector(selector)
 	addedLabels = []string{
 		"!testDoesNotExist",
 	}
-	addedKeys = []string{
-		"!testDoesNotExist",
-	}
-	addedVals = []string{
-		"",
-	}
+	addedVals = map[string][]string{}
 	expectedLabels = append(expectedLabels, addedLabels...)
-	expectedKeys = append(expectedKeys, addedKeys...)
-	expectedVals = append(expectedVals, addedVals...)
+	for k, v := range addedVals {
+		expectedVals[k] = append(expectedVals[k], v...)
+	}
 
 	if len(labels) != len(expectedLabels) {
 		t.Errorf("TestparseSelector failed @ labels length comparison")
 	}
 
-	if len(keys) != len(expectedKeys) {
-		t.Errorf("TestparseSelector failed @ keys length comparison")
-	}
-
-	if len(vals) != len(vals) {
+	if len(vals) != len(expectedVals) {
 		t.Errorf("TestparseSelector failed @ vals length comparison")
 	}
 
 	if !reflect.DeepEqual(labels, expectedLabels) {
 		t.Errorf("TestparseSelector failed @ label comparison")
 	}
 
-	if !reflect.DeepEqual(keys, expectedKeys) {
-		t.Errorf("TestparseSelector failed @ key comparison")
-	}
-
 	if !reflect.DeepEqual(vals, expectedVals) {
 		t.Errorf("TestparseSelector failed @ value comparison")
 	}
 }
+func TestFlattenNameSpaceSelector(t *testing.T) {
+	firstSelector := &metav1.LabelSelector{
+		MatchExpressions: []metav1.LabelSelectorRequirement{
+			metav1.LabelSelectorRequirement{
+				Key:      "testIn",
+				Operator: metav1.LabelSelectorOpIn,
+				Values: []string{
+					"backend",
+				},
+			},
+			metav1.LabelSelectorRequirement{
+				Key:      "pod",
+				Operator: metav1.LabelSelectorOpIn,
+				Values: []string{
+					"a",
+				},
+			},
+			metav1.LabelSelectorRequirement{
+				Key:      "testExists",
+				Operator: metav1.LabelSelectorOpExists,
+				Values:   []string{},
+			},
+			metav1.LabelSelectorRequirement{
+				Key:      "ns",
+				Operator: metav1.LabelSelectorOpIn,
+				Values: []string{
+					"t",
+				},
+			},
+		},
+		MatchLabels: map[string]string{
+			"c": "d",
+			"a": "b",
+		},
+	}
+
+	testSelectors := FlattenNameSpaceSelector(firstSelector)
```

Comment:
Will be very helpful if there are comparison between expected values and testSelectors to easily understand what expected values are.

Reply:
Agreed, added comparison values